reroll-initiative/reroll-initiative.js

```diff
@@ -0,0 +1,203 @@
+            default: RRI_CONFIG.actorTypes,
+            //default: "",
+            type: String,
+            scope: "world",
+            onChange: setting => {
+              console.log("setting change",setting);
+            }
+        });
+        */
+    }
+
+    /**
+     * Returns the default module settings
+     */
+    _defaultSettings() {
+        this.settings = RRI_SETTINGS;
+        console.log("Restting reroll-initiative settings to defaults:",RRI_DEFAULT_SETTINGS);
+    }
+
+    /**
+     * Saves current class instance settings back to game settings
+     */
+    _saveSettings () {
+        game.settings.set("reroll-initiative","rriSettings",this.settings);
+    }
+
+    /**
+     * Loads current class instance settings from game settings
+     */
+    _loadSettings (){
+        let settings = game.settings.get("reroll-initiative","rriSettings");
+        this.settings = settings;
+        console.log(this.settings);
+    }
+    
+    /**
+     * Get the current class instance settings (for external use)
+     */
+    get settings() {
+        this._loadSettings();
+        return this.settings;
+    }
+
+    /**
+     * Change the current class instance settings (for external use)
+     * @param {Object} incomingSettings
+     */
+    set settings(incomingSettings) {
+        this.settings = incomingSettings;
+        return this._saveSettings();
+    }
+
+    /**
+     * @name postUpdateCombatHook
+     * @description Hook on combat update and if round in update is greater than previous -- call resetAndReroll
+     */
+    postUpdateCombatHook() {
+        Hooks.on("updateCombat", (combat,update) =>  {
+            this._loadSettings();
+
+            if(this.settings.reroll){
+                
+                if(update.round && combat.previous && update.round > combat.previous.round){
+                    //console.log("Reroll-Initiative: Round incremented - rerolling initiative")
+                    this.resetAndReroll(combat);
+                }
+            }
+            
+        }); 
+    }
+
+    /**
+     * @name resetAndReroll
+     * @param {Combat} combat
+     * @description For the given combat instance, call the resetAll method and the rollAll method
+     */
+    async resetAndReroll(combat){
+        await combat.resetAll();
+        combat.rollAll();
+    }
+}
+
+/**
+ * @class RerollInitiativeConfig
+ * @description Handles the configuration of the module
+ * @
+ */
+class RerollInitiativeConfig {
+    constructor(){
+        this._hookRenderCombatTrackerConfig();
+        this.rri = game["reroll-initiative"]["rri"];
+    }    
+
+    _hookRenderCombatTrackerConfig(){
+        Hooks.on("renderCombatTrackerConfig", (app, html) => {
+            console.log(this.rri);
+            let settings = this.rri.settings;
+            console.log(settings);
+            let reroll = settings.reroll;
+            console.log(reroll);
+
+            let submit = html.find('button[type="submit"]');
```

Comment:
I need to read move into this, it seems very error prone to do a lookup like this and it by id. Your scope is very large and you could pick up another button on the page

Reply:
in this case the scope is restricted to just the combat config form. there might be a better way but this suggestion came from atropos